File: back/src/main/java/com/triptrace/domain/image/image/exception/ImageProcessException.java
package com.triptrace.domain.image.image.exception;

import com.triptrace.global.error.ErrorCode;

public class ImageProcessException extends RuntimeException {
    private final String resultCode;
    private final String msg;
    public ImageProcessException(String resultCode, String message, Throwable cause) {
        super(message, cause);
        this.resultCode = resultCode;
        this.msg = message;
    }
    public ImageProcessException(String resultCode, String message) {
        this(resultCode, message, null);
    }

    public ImageProcessException(ErrorCode errorCode, String message) {
        super(message);
        this.resultCode = "%s-%s".formatted(errorCode.getCode(), errorCode.getDomain().getCode());
        this.msg = message;
    }
    public ImageProcessException(ErrorCode errorCode){
        this(errorCode, errorCode.getMessage());
    }

    public String getResultCode() {
        return resultCode;
    }

    public String getMsg() {
        return msg;
    }
}

File: back/src/main/java/com/triptrace/domain/image/image/exception/ImageProcessException.kt
package com.triptrace.domain.image.image.exception

import com.triptrace.global.error.ErrorCode

class ImageProcessException : RuntimeException {
    val resultCode: String
    val msg: String
    constructor(resultCode: String, message: String, cause: Throwable? = null) : super(message, cause) { this.resultCode = resultCode; msg = message }
    @JvmOverloads constructor(errorCode: ErrorCode, message: String = errorCode.message) : super(message) { resultCode = "${errorCode.code}-${errorCode.domain.code}"; msg = message }
}
